public namespace uf;
public import common -;
public import io::errno -;

public union foo {
  public errno_t err;
  public string value;
}

{
  exception::set_stack_trace_limit(0);
  try {
    foo x;
    x.err = EINTR;
    println(x.value); // throws
  } catch (c_exception ex) {
    println("caught", ex);
  }
}

